core/vm/interpreter.go

```diff
@@ -228,7 +234,7 @@ func (in *Interpreter) Run(snapshot int, contract *Contract, input []byte) (ret
 		// if the operation returned a value make sure that is also set
 		// the last return data.
 		if res != nil {
```

Comment:
References: https://github.com/ethereum/EIPs/pull/211#issuecomment-287728214

> To summarize: Any opcode that attempts to create a new call stack frame resets the buffer of the current stack frame right before it executes, even if that opcode fails.

Also, from the EIP itself -
 https://github.com/ethereum/EIPs/blob/aeaffcca26fa688f5015da7ce059c886e90ef374/EIPS/returndatacopy.md:

> As an optimization, it is possible to share the return data buffer across call frames because only one will be non-empty at any time.